bundles/org.eclipse.jface/src/org/eclipse/jface/resource/URLImageDescriptor.java

```diff
@@ -249,8 +249,12 @@ private static String getxPath(String name, int zoom) {
 	private static String getFilePath(URL url, boolean logIOException) {
 		try {
 			if (!InternalPolicy.OSGI_AVAILABLE) {
-				if (FILE_PROTOCOL.equalsIgnoreCase(url.getProtocol()))
-					return IPath.fromOSString(url.getFile()).toOSString();
+				if (FILE_PROTOCOL.equalsIgnoreCase(url.getProtocol())) {
+					String filePath = IPath.fromOSString(url.getFile()).toOSString();
```

Comment:
can't we use `Path.of(...)` directly.

Reply:
I think `Path.of(url.toURI()).toString()` is even less readable than `IPath.fromOSString(url.getFile()).toOSString()`, in addition to also having to deal with the URISyntaxException.

Reply:
> `IPath.fromOSString(url.getFile()).toOSString()`

to be honest this looks really bogus, why not simply 
```suggestion
					String filePath = url.getFile();
```

but I more was about that below you are using a path anyways so


```
p = Path.of(url.toURI())
if (Files.exists(Path.of(p))) {
  return p.toString();
}
```

Reply:
> why not simply

As mentioned, you now have to also worry about the potential URISyntaxException and are now depending on something ambiguous as a `toString()` method...

But as long as the method behaves consistently, I don't really mind either way.

Reply:
What now raises the question if we should not extract it to an own method, and why we need an existence checks for files but not for regular URLs ... so maybe the caller will check for the existence anyways later on?

Reply:
> What now raises the question if we should not extract it to an own method, and why we need an existence checks for files but not for regular URLs

Because URLs are not used in combination with the `ImageFileNameProvider`. The reason while files get a special treatment is primarily performance and I don't think extracting this logic to a separate method only adds unnecessary complexity.

> so maybe the caller will check for the existence anyways later on?

The caller will not. Otherwise this issue wouldn't even exist in the first place. And this is also not expected, as specified in the JavaDoc;

https://github.com/eclipse-platform/eclipse.platform.ui/blob/789b373c50e3c665374f58c1427cb1fdd2b53b2b/bundles/org.eclipse.jface/src/org/eclipse/jface/resource/URLImageDescriptor.java#L247

Reply:
>  I don't think extracting this logic to a separate method only adds unnecessary complexity.

At least you would not need to adjust the logic in two places now ...

> as specified in the JavaDoc;

If I read it correctly `FileNameProvider` it should only return `null` if the zoom != 100 .. but this seems flawed before.

Reply:
> At least you would not need to adjust the logic in two places now ...

Updated.

> If I read it correctly FileNameProvider it should only return null if the zoom != 100 .. but this seems flawed before.

That is true and probably also why the IOException is explicitly logged in such a case. But that's something that goes beyond the scope of this PR.

Reply:
@laeubi I've moved the conversion from URL to String to a separate method. Do you see anything else I've missed? Otherwise I'd merge this within the next days.